code/modules/mob/living/carbon/xenomorph/castes/hivemind/abilities_hivemind.dm

```diff
@@ -1 +1,243 @@
+
+	var/decoys
+	var/area/decoy_area
+	while(decoys < HIVEMIND_REPOSITION_CORE_DECOY_NUMBER) //Populate our list of areas
+		decoy_area = pick(decoy_area_list) //Pick random area for our  decoy.
+		decoy_area_list -= decoy_area //Remove it from our list of possible decoy options
+		buffer_list += sanitize(decoy_area.name)
+		++decoys
+
+	var/buffer_list_pick
+	while(buffer_list.len > 0) //Now populate our randomized order list for the announcement
+		buffer_list_pick = pick(buffer_list) //Get random entry in the list
+		buffer_list -= buffer_list_pick //Remove that random entry from the buffer
+		if(buffer_list.len > 0) //Add that random entry to the final list of areas
+			details += ("[buffer_list_pick], ")
+		else
+			details += ("[buffer_list_pick].")
+
+	priority_announce("Attention: Anomalous energy readings detected in the following areas: [details.Join(" ")] Further investigation advised.", "Priority Alert", sound = 'sound/AI/commandreport.ogg')
+
+
+/datum/action/xeno_action/activable/reposition_core/proc/check_build_location(turf/target_turf, mob/living/carbon/xenomorph/hivemind/X)
+
+	if(!X || !target_turf) //Sanity
+		return FALSE
+
+	if(target_turf.z != X.core.z)
+		to_chat(X, "<span class='xenodanger'>We cannot transfer our core here.</span>")
+		return FALSE
+
+	if(target_turf.density) //Check to see if there's room
+		to_chat(X, "<span class='xenodanger'>We require adequate room to transfer our core.</span>")
+		return FALSE
+
+	var/obj/effect/alien/weeds/W = locate() in range(0, target_turf) //Make sure we actually have weeds at our destination.
+	if(!W)
+		to_chat(X, "<span class='xenodanger'>There are no weeds for us to transfer our consciousness to!</span>")
+		return FALSE
+
+	for(var/obj/structure/O in target_turf.contents)
+		if(O.density && !(O.flags_atom = ON_BORDER))
+			to_chat(X, "<span class='xenodanger'>An object is occupying this space!</span>")
+			return FALSE
+
+	return TRUE
+
+
+
+// ***************************************
+// *********** Mind Wrack
+// ***************************************
+/datum/action/xeno_action/activable/mind_wrack
+	name = "Mind Wrack"
+	action_icon_state = "screech"
+	mechanics_text = "Afflicts the target's mind with disorienting and painful psychic energy. Effect is greater the closer the target is to your core, and the more xenos that are within 5 tiles when used. Must have a core or other xeno within 5 tiles to use."
+	ability_name = "mind wrack"
+	plasma_cost = 100
+	keybind_signal = COMSIG_XENOABILITY_MIND_WRACK
+	cooldown_timer = 30 SECONDS
+
+/datum/action/xeno_action/activable/mind_wrack/on_cooldown_finish()
+	owner.playsound_local(owner.loc, 'sound/voice/alien_drool1.ogg', 50, TRUE)
+	to_chat(owner, "<span class='xenonotice'>We regain the strength to assault the minds of our enemies.</span>")
+	return ..()
+
+/datum/action/xeno_action/activable/mind_wrack/can_use_ability(atom/target, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
```

Comment:
Include a check for 'near weeds' similar to `plant_weeds` in this please.

Reply:
Or better yet, convert the existing 'near weeds' checks into their own proc to cut down on code duplication.

Reply:
Is a weed check necessary? If you're not near weeds you go back to Brazil anyways, whether or not you remain still.

Reply:
Because I'm asking you to remove the weed check from `life()`

Reply:
This is done.